interfaces/snap_app_set.go

```diff
@@ -149,6 +149,45 @@ func (a *SnapAppSet) Runnables() []snap.Runnable {
 	return runnables
 }
 
+// PlugRunnables returns a list of all runnables that should be connected to the
+// given plug.
+func (a *SnapAppSet) PlugRunnables(plug *ConnectedPlug) []snap.Runnable {
+	apps := a.info.AppsForPlug(plug.plugInfo)
+	hooks := a.info.HooksForPlug(plug.plugInfo)
+	for _, component := range a.components {
+		hooks = append(hooks, component.HooksForPlug(plug.plugInfo)...)
+	}
+
+	runnables := make([]snap.Runnable, 0, len(apps)+len(hooks))
+	for _, app := range apps {
+		runnables = append(runnables, snap.AppRunnable(app))
+	}
+
+	for _, hook := range hooks {
+		runnables = append(runnables, snap.HookRunnable(hook))
+	}
+
+	return runnables
+}
+
+func (a *SnapAppSet) SlotRunnables(slot *ConnectedSlot) []snap.Runnable {
+	apps := a.info.AppsForSlot(slot.slotInfo)
+	hooks := a.info.HooksForSlot(slot.slotInfo)
+
+	// TODO: if components ever get slots, they will need to be considered here
+
+	runnables := make([]snap.Runnable, 0, len(apps)+len(hooks))
+	for _, app := range apps {
+		runnables = append(runnables, snap.AppRunnable(app))
+	}
+
+	for _, hook := range hooks {
+		runnables = append(runnables, snap.HookRunnable(hook))
+	}
+
+	return runnables
```

Comment:
Possibly this code snippet can be an ancillary method shared with `PlugRunnables`